src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -94,7 +104,12 @@ public static GitHubDataFetcher instance() {
    * @throws IOException occurred during REST call to GitHub API.
    */
   private List<GHCommit> commitsFor(GHRepository repository) throws IOException {
-    return repository.listCommits().toList();
+    try {
+      return repository.listCommits().toList();
+    } catch (HttpException e) {
+      LOGGER.error(String.format("Could not fetch commits from %s", repository.getUrl()), e);
+      return Collections.emptyList();
```

Comment:
Should this throw an IOException? after catching HttpException?

Reply:
If a repository is empty, `listCommits()` throws an exception. If it's called in a data provider, then it fails. So, there should be a check somewhere. I don't think that returning an empty list makes any harm at the moment.